File: apps/mybookkeeper/backend/app/schemas/leases/generate_defaults_response.py
"""Response schema for GET /lease-templates/{id}/generate-defaults.

Returns the resolved default value and provenance for each placeholder that
has a ``default_source`` spec. Placeholders without a ``default_source`` are
omitted — they're manual-entry only.
"""
from __future__ import annotations

from pydantic import BaseModel, ConfigDict


class PlaceholderDefault(BaseModel):
    """Resolved default for a single placeholder key."""

    key: str
    value: str | None
    # "applicant" | "inquiry" | "today" | None (when nothing resolved)
    provenance: str | None

    model_config = ConfigDict(from_attributes=False)


class GenerateDefaultsResponse(BaseModel):
    """Resolved defaults for all placeholders in a template."""

    defaults: list[PlaceholderDefault]

    model_config = ConfigDict(from_attributes=False)

File: apps/mybookkeeper/frontend/src/shared/types/lease/generate-defaults-response.ts
import type { PlaceholderProvenance } from "@/shared/types/lease/placeholder-provenance";

/**
 * Resolved default for a single placeholder key.
 * Mirrors backend ``PlaceholderDefault``.
 */
export interface PlaceholderDefault {
  key: string;
  value: string | null;
  provenance: PlaceholderProvenance;
}

/**
 * Response from ``GET /lease-templates/{id}/generate-defaults``.
 * Mirrors backend ``GenerateDefaultsResponse``.
 */
export interface GenerateDefaultsResponse {
  defaults: PlaceholderDefault[];
}
